### PR TITLE
Push v24.1.3 into main

### PR DESCRIPTION
**Pull request by ridz1208**
DO NOT SQUASH

Conflicts:
```
	both modified:   Makefile
	both modified:   modules/electrophysiology_browser/README.md
	both added:      modules/electrophysiology_browser/jsx/react-series-data-viewer/package-lock.json
	both modified:   modules/imaging_browser/templates/form_viewSession.tpl
	both added:      package-lock.json
	both modified:   package.json
	both modified:   php/libraries/Module.class.inc
	both modified:   php/libraries/NDB_BVL_Feedback.class.inc
	both modified:   php/libraries/NDB_BVL_Instrument.class.inc
	both modified:   src/Http/Endpoint.php
	both modified:   src/Router/BaseRouter.php
	both modified:   tools/single_use/Cleanup_multiple_firstVisits.php
	both modified:   tools/single_use/Remove_Data_entry_completion_status_instr_column.php
	both modified:   tools/single_use/Set_Required_elements_completed_flag.php
	both modified:   tools/single_use/populate_visits.php
	both modified:   webpack.config.js
```

NOTES:

 - changes to the following files should be reviewed carefully as the chronological order I use to merge the changes is not necessarily the correct one here
   - Makefile
   - package.json and package-lock.json
   - src/Router/BaseRouter.php
 - everything else should still be reviewed
 - **Make DOES NOT run to completion on this branch** I was not able to fix it and I figured someone can take a look at this to see if its an issue with my environment of the branch is broken @driusan @laemtl

### CODE REVIEW

File: .github/workflows/loristest.yml [hunk 1]
```diff
@@ -71,6 +71,11 @@ jobs:
       if: steps.composer-cache.outputs.cache-hit != 'true'
       run: composer install --prefer-dist --no-progress --no-suggest
 
+    - name: Setup Node.js
```
Comment (laemtl) [inline .github/workflows/loristest.yml:74]: Is this required in main ? I don't think we have any issues with the tests.

**Comment by laemtl** [comment]
@ridz1208 #8483 fixes your issue

File: .github/workflows/loristest.yml (hunk 1 above)
Comment (kongtiaowang) [reply .github/workflows/loristest.yml:74]: This section solely addresses the correction of the node version in the GitAction testing environment. It is a prerequisite for the main functionality, despite not having made any changes to the Loris core code.

File: Makefile [hunk 2]
```diff
@@ -6,13 +6,15 @@ all: VERSION
...
 # If anything changes, re-generate the VERSION file
-VERSION: .
+VERSION:
 	tools/gen-version.sh
 
 phpdev:
 	composer install
 
-dev: VERSION phpdev
+dev:
```
Comment (laemtl) [inline Makefile:15]: Can you change for 
```
dev: VERSION phpdev
	npm ci
	npm run compile
```

Comment (ridz1208) [reply Makefile]: fixed

File: .github/workflows/loristest.yml (hunk 1 above)
Comment (ridz1208) [reply .github/workflows/loristest.yml:74]: should I just delete that one row ?

Comment (laemtl) [reply .github/workflows/loristest.yml:74]: So node v18 is not supported if I understand well? Should it be specified somewhere in the install notes?

Comment (ridz1208) [reply .github/workflows/loristest.yml:74]: someone should verify I guess?

Comment (laemtl) [reply .github/workflows/loristest.yml:74]: Yes, I did. See my edit above.

Comment (kongtiaowang) [reply .github/workflows/loristest.yml:74]: Please delete them. The git action appears to be functioning properly.

File: Makefile (hunk 2 above)
Comment (ridz1208) [reply Makefile]: done

**Comment by laemtl** [comment]
The EEG Readme instructions were overridden by the merge, I fixed it in #8542

**Pull request merged**